src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/FermiLiteAssemblyHandler.java

```diff
@@ -0,0 +1,73 @@
+package org.broadinstitute.hellbender.tools.spark.sv.evidence;
+
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.SVFastqUtils;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.SVUtils;
+import org.broadinstitute.hellbender.utils.bwa.BwaMemAligner;
+import org.broadinstitute.hellbender.utils.bwa.BwaMemAlignment;
+import org.broadinstitute.hellbender.utils.bwa.BwaMemIndexCache;
+import org.broadinstitute.hellbender.utils.fermi.FermiLiteAssembler;
+import org.broadinstitute.hellbender.utils.fermi.FermiLiteAssembly;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/** This LocalAssemblyHandler aligns assembly contigs with BWA, along with some optional writing of intermediate results. */
+public final class FermiLiteAssemblyHandler implements FindBreakpointEvidenceSpark.LocalAssemblyHandler {
+    private static final long serialVersionUID = 1L;
+    private final String alignerIndexFile;
+    private final int maxFastqSize;
+    private final String fastqDir;
+    private final boolean writeGFAs;
+
+    public FermiLiteAssemblyHandler( final String alignerIndexFile, final int maxFastqSize,
+                                     final String fastqDir, final boolean writeGFAs ) {
+        this.alignerIndexFile = alignerIndexFile;
+        this.maxFastqSize = maxFastqSize;
+        this.fastqDir = fastqDir;
+        this.writeGFAs = writeGFAs;
+    }
+
+    @Override
+    public AlignedAssemblyOrExcuse apply( final Tuple2<Integer, List<SVFastqUtils.FastqRead>> intervalAndReads ) {
+        final int intervalID = intervalAndReads._1();
+        final List<SVFastqUtils.FastqRead> readsList = intervalAndReads._2();
+
+        final int fastqSize = readsList.stream().mapToInt(FastqRead -> FastqRead.getBases().length).sum();
+        if (fastqSize > maxFastqSize) {
+            return new AlignedAssemblyOrExcuse(intervalID, "no assembly -- too big (" + fastqSize + " bytes).");
+        }
+
+        final String fastqName = String.format("%s/%s.fastq",fastqDir, AlignedAssemblyOrExcuse.formatAssemblyID(intervalID));
```

Comment:
@cwhelan I can restore this argument as optional, but:
1) It isn't optional in our launch script -- we always specify it.
2) I thought that Valentin's genotyper relied on the fastq's, and so we'd always want to produce them.
Would you prefer to retain it as optional?

Reply:
Those are both valid points.. I just wasn't sure why the behavior was being changed here. Since we don't really know the performance characteristics of @vruano's genotyper yet (and depending on how that goes we might need to look for a more efficient solution to data transfer) I thought we might want to keep the flexibility for now. If it's not a pain to revert I think I'd rather keep it optional.

Reply:
roger